lib/biran/configurinator.rb

```diff
@@ -17,12 +17,11 @@ def configure
       def env= env
         @env = env
       end
+    end
 
-      def env
-        return @end if @env
-        return Rails.env if defined? Rails
-        DEFAULT_ENV
-      end
+    def env
+      return @env if @env
+      @env = app_env
```

Comment:
This stuff here and above with the `env=` method is where I am still not sure its being done properly for what I am looking for. As of now, this works as expected when passing in environment variables and with using a default value of `development` if nothing is supplied. I just don't think the "setter" part is working as expected.
In the rake task, I am trying something like the following to test:
```
namespace :config do
  config = Biran::Configurinator.new
  config.env = 'staging'
  puts config.env
```
and its not working. I have tried variations of the above, but all end up with roughly the same error of
`NoMethodError: undefined method `env=' for #<Biran::Configurinator:0x00007fd3f2a00af0>`
So I am doing this wrong OR the methods are not right.
I think that is the last thing to have this working like I would like to see.

Reply:
Should it move to part of the initializer or something else?

Reply:
We need to move app_env to be set on

https://github.com/amco/biran/pull/41/files#diff-08222fcadd2b9def888351b5d58d6611R17

Reply:
We don't want it set under the app settings. At least I don't think so. That block gets filtered out when passed to the ERB object and would become unavailable. Its set in the hash here - https://github.com/amco/biran/pull/41/files#diff-213904ac166ac8b8343fa4c1ab0dece8R63. Maybe I am missing your intention.

Reply:
We can add it to the allowed properties, so that is not a stopper... but let me check that line you posted

Reply:
@javierg if you have a chance, I would like to revisit this. I have two non-rails apps waiting on this issue to be fixed. I think its just the setter stuff. I think I am blocked at the moment due to lack of understanding. Or @seancookr if you have any ideas....

Reply:
for some reason the setter of `env` is set as class method instead of instance method... @brlanier was this on purpose?

Reply:
I have no idea @javierg that is why I was asking. Its been a long time, so I would have to go back and see where I was at. I was struggling to understand what I was even doing.

Reply:
I see... so, you can do this:

```
    def env
      return @env if @env # memoized value for env
      @env = self.class.env || app_env # checks for class level set env if not use the one from defaults
    end
```

Also you will need to have a instance setter for env

```
def env= value
  self.class.env = value
end
```

Trying to figure it out why the accessor is in the class level